resources/remote_management.rb

```diff
@@ -3,24 +3,85 @@
 provides :remote_management
 default_action :enable
 
-kickstart = '/System/Library/CoreServices/RemoteManagement/ARDAgent.app/Contents/Resources/kickstart'
+property :users,
+         [String, Array],
+         default: 'all',
+         description: 'The user(s) whose ARD privileges will be configured.',
+         coerce: ->(p) { [p].flatten }
+
+property :privileges,
+         [String, Array, RemoteManagement::Privileges::Mask],
+         default: 'all',
+         description: 'The desired privileges to bestow upon the given users.',
+         coerce: ->(p) { p.is_a?(RemoteManagement::Privileges::Mask) ? p : RemoteManagement::Privileges::Mask.new(privileges: p) }
+
+property :computer_info,
+         [String, Array],
+         default: [],
+         description: 'Info fields; helpful for stratifying computers in ARD client app.',
+         coerce: ->(p) { p.is_a?(Array) ? p.compact.map(&:to_s) : [p] },
+         callbacks: { 'has too many elements; computer info excepts up to four info fields' => ->(p) { (p.is_a?(Array) && p.size < 4) } }
+
+load_current_value do |desired|
+  current_value_does_not_exist! unless RemoteManagement.activated? && RemoteManagement.current_users_configured?(desired.users) && RemoteManagement.current_users_have_identical_masks?(desired.users)
+  privileges RemoteManagement.current_mask(desired.users)
+  computer_info RemoteManagement.current_computer_info
+end
+
+# TODO; the enable action should be decoupled from configuration; configure action should be added; default action should be [:configure, :enable]
```

Comment:
@jazaval what do you think about splitting the `:enable` action into two actions `[:configure, :enable]`. `:enable` is a bit misleading as a lot of configuration is done.